### ISSUE TITLE
Addition of siamese Contrastive loss function

### ISSUE THREAD

**Issue by arcAman07** [tensorflow/similarity#248]
Addition of New Contastive Loss function which is used as a primary metric to calculate loss in siamese networks. If it needs to be added, I would love to work on the PR.
Reference : http://yann.lecun.com/exdb/publis/pdf/hadsell-chopra-lecun-06.pdf

**Comment by owenvallis** [comment]
Thanks for offering to work on the PR! This loss is a little older and it comes before Triplet-Loss and the more recent variations on other metric losses. What would you see as the primary use case for the loss, e.g., providing a baseline for benchmarking or does this loss outperform other losses on certain problems?

**Comment by arcAman07** [comment]
Personally haven't benchmarked the triplet loss and siamese contrastive loss functions for different datasets, but personally siamese contrastive loss function has always yielded me good results in all my projects where I am training siamese networks. It is also a pretty popular function in the ML community as it an add-on on keras and a PR for its inclusion is on its way even in pytorch. I am of the opinion as this research paper is so important, along with this loss function does yield amazing results ( it also offers diversity in loss functions ), it will be a good addition to this library.

**Comment by owenvallis** [comment]
Sounds good, thanks. Let's add the PR to the development branch and then we can merge it in to master along with the other recent updates.

**Comment by arcAman07** [comment]
Great would start working pn the PR right away 👍
